include/aklite.h

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2021 Foundries.io
+// SPDX-License-Identifier: Apache-2.0
+
+#include <string>
+
+#include <boost/filesystem.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include "json/json.h"
+
+class LiteClient;
+
+/**
+ * A high-level representation of a TUF Target in terms applicable to a
+ * FoundriesFactory.
+ */
+class TufTarget {
+ public:
+  TufTarget(std::string name, std::string sha256, int version, Json::Value custom)
+      : name_(std::move(name)), sha256_(std::move(sha256)), version_(version), custom_(std::move(custom)) {}
+
+  /**
+   * Return the TUF Target name. This is the key in the targets.json key/value
+   * singed.metadata dictionary.
+   */
+  std::string Name() const { return name_; }
+  /**
+   * Return the sha256 OStree hash of the Target.
+   */
+  std::string Sha256Hash() const { return sha256_; }
+  /**
+   * Return the FoundriesFactory CI build number or in TUF, custom.version.
+   */
+  int Version() const { return version_; }
+
+  /**
+   * Return TUF custom data for a Target.
+   */
+  Json::Value Custom() const { return custom_; }
+
+ private:
+  std::string name_;
+  std::string sha256_;
+  int version_;
+  Json::Value custom_;
+};
+
+/**
+ * The response from an AkliteClient call to CheckIn()
+ */
+class CheckInResult {
+ public:
+  enum class Status {
+    Ok,        // check-in was good
+    OkCached,  // check-in failed, but locally cached meta-data is still valid
+    Failed,    // check-in failed and there's no valid local meta-data
+  };
+  CheckInResult(Status status) : status(status) {}
+  Status status;
+  std::vector<TufTarget> Targets() const { return targets_; }
+  TufTarget GetLatest() const { return targets_.back(); }
+
+ private:
+  friend class AkliteClient;
+  std::vector<TufTarget> targets_;
+};
+
+/**
+ * The response from an AkliteClient call to Download()
+ */
+class DownloadResult {
+ public:
+  enum class Status {
+    Ok,
+    DownloadFailed,
+    VerificationFailed,
+  };
+  DownloadResult(Status status, std::string description = "") : status(status), description(std::move(description)) {}
```

Comment:
I think the ctor could be redundant in this case, something like `DownloadResult res{<status>, <descr>}` or `return DownloadResult{<status>, <descr>}` should work.

Reply:
179aa36 (HEAD -> libaklite, origin/libaklite) fixup! api: Add Install method
beb11cd fixup! api: Add Download method